File: assignment/269p3-2.R
pima <- read.csv('./data/pima2.csv', header = T)
par(mfrow = c(2, 1))
for (col in c('glucose', 'pressure', 'triceps', 'insulin', 'mass', 'pedigree', 'age')) {
    boxplot(get(col) ~ diabetes, data = pima, main = paste0(col, ' boxplot'), col = 1:2)
    hist_pos <- hist(pima[[col]][pima$diabetes == 'pos'], plot = F)
    hist_neg <- hist(pima[[col]][pima$diabetes == 'neg'], plot = F)
    plot(hist_pos,
         col = adjustcolor("red", alpha = 0.5),
         ann = FALSE,
         ylim = c(0, max(hist_pos$counts, hist_neg$counts)))
    plot(hist_neg,
         col = adjustcolor("blue", alpha = 0.5),
         add = TRUE,
         ylim = c(0, max(hist_pos$counts, hist_neg$counts)))
    title(main = paste0(col, ' histogram'))
}

File: assignment/269p3-2.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

pima = pd.read_csv('../data/pima2.csv')

describe = pima.groupby('diabetes').describe()

for col in ['glucose', 'pressure', 'triceps', 'insulin', 'mass', 'pedigree', 'age']:
    print('#' * 10 + ' ' + col)
    print(describe[col])
    sns.set(style='whitegrid')
    sns.boxplot(x='diabetes', y=col, data=pima)
    plt.gca().set(title='%s\'s boxplot' % (col))
    plt.show()
    pos_total = pima.loc[pima.diabetes == 'pos'][col]
    neg_total = pima.loc[pima.diabetes == 'neg'][col]
    args = dict(alpha=0.5, bins=10)
    plt.hist(pos_total, **args, color='b', label='pos')
    plt.hist(neg_total, **args, color='r', label='neg')
    plt.gca().set(title='%s\'s histogram' % (col))
    plt.legend()
    plt.show()
